### PR TITLE
Add Static Game Data

### PR DESCRIPTION
**Pull request by kuttnig**
- convert json to java object (and vice versa)
- add some dummy json files for testing
- add property to class Card (which stores the resource name of the related img)

### CODE REVIEW

**Comment by raphaelseher** [comment]
> There is no coverage information present for the Files changed

|Total Project Coverage|23.04%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [48.1%]|:green_apple:|
|:-|:-:|:-:|
|[CardAction.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FCardAction.java)|97.12%|:green_apple:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|81.38%|:green_apple:|
|[DeployParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FDeployParams.java)|55.81%|:green_apple:|
|[FogParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FFogParams.java)|47.37%|:green_apple:|
|[Card.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCard.java)|43.75%|:green_apple:|
|[DamageParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FDamageParams.java)|33.33%|:green_apple:|
|[DetailedCardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardViewModel.java)|0%|:green_apple:|
|[JsonConverter.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2FJsonConverter.java)|0%|:green_apple:|
|[HealParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FHealParams.java)|0%|:green_apple:|
|[BoostParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FBoostParams.java)|0%|:green_apple:|
|[SwapParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FSwapParams.java)|0%|:green_apple:|
|[OrderParams.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2FOrderParams.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/322186cf8441ded5ff320d37cc53b064c6802c01/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|

|Total Project Coverage|27.7%|:green_apple:|
|:-|:-:|:-:|

**Comment by kuttnig** [comment]
@raphaelseher so würde die neue Struktur von CardAction aussehen. Wenn du Lust hast schau bitte mal drüber. Ich würde dann darauf aufbauend die Json-Files erstellen.
Ps: Nur nochmal zur Sicherheit: PR nach dem Review bitte nicht schließen.
Lg

**Comment by raphaelseher** [comment]
|File|Coverage [44.1%]|:green_apple:|
|:-|:-:|:-:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|81.15%|:green_apple:|
|[DeployTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FDeployTrigger.java)|66.67%|:green_apple:|
|[OrderTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FOrderTrigger.java)|52.94%|:green_apple:|
|[Card.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCard.java)|45.61%|:green_apple:|
|[JsonGenerator.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonGenerator.java)|0%|:green_apple:|
|[JsonConverter.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonConverter.java)|0%|:green_apple:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|0%|:green_apple:|
|[TargetRowAction.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetRowAction.java)|0%|:green_apple:|
|[TargetUnitAction.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetUnitAction.java)|0%|:green_apple:|
|[CardAction.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FCardAction.java)|0%|:green_apple:|
|[DetailedCardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardViewModel.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/cf588f55c3ee269492b5999ef3097c1a3694af13/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|

|Total Project Coverage|25.15%|:green_apple:|
|:-|:-:|:-:|

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/TargetRowAction.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package at.aau.se2.gamelogic.models.cardactions.actions;
+
+import java.util.ArrayList;
+
+public class TargetRowAction {
+  private int targetingCardUUID;
+  private ArrayList<Integer> targetedRowsUUIDs;
+  private final boolean hasRandomTargets;
+  private final boolean getsRandomStatus;
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/TargetRowAction.java:9]: Beim Naming machen die meisten:

randomTargets & randomStatus.

Und die getter heißen dann hasRandomTargets & hasRandomStatus.

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/TargetUnitAction.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    BOOST,
+    SWAP;
+  }
+
+  private int targetingCardUUID;
+  private ArrayList<Integer> targetedCardsUUIDs;
+  private int points;
+  private boolean hasRandomTargets;
+  private boolean canTargetAlliedUnits;
+  private boolean canTargetEnemyUnits;
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/TargetUnitAction.java:25]: Hier auch Naming anpassen.

Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/TargetUnitAction.java:25]: Hier auch Naming anpassen.

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/CardAction.java [hunk 3]
```diff
@@ -0,0 +1,3 @@
+package at.aau.se2.gamelogic.models.cardactions.actions;
+
+public interface CardAction {}
```
Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/CardAction.java:3]: Wir haben jetzt ein Interface und ein Klasse names CardAction.


Das Interface wird aber leer implementiert, also eigentlich obsolete?

Comment (raphaelseher) [inline gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/CardAction.java:3]: Wir haben jetzt ein Interface und ein Klasse names CardAction.


Das Interface wird aber leer implementiert, also eigentlich obsolete?

File: gamelogic/src/test/java/at/aau/se2/gamelogic/GameLogicTest.java [hunk 4]
```diff
@@ -66,10 +61,11 @@ public void setup() {
     mockConnector = mock(FirebaseConnector.class);
     mockGameStateMachine = mock(GameStateMachine.class);
     sut = new GameLogic(mockConnector, mockGameStateMachine);
-    mockCallback = mock(CardActionCallback.class);
-    sut.registerCardActionCallback(mockCallback);
+    // mockCallback = mock(CardActionCallback.class);
+    // sut.registerCardActionCallback(mockCallback);
   }
 
+  /*
```
Comment (raphaelseher) [inline gamelogic/src/test/java/at/aau/se2/gamelogic/GameLogicTest.java:68]: Tests die wir nicht mehr brauchen können raus.

Machen wir neu, wenn wir die neuen Triggers/Actions in die Gamelogic implementieren.

Comment (raphaelseher) [inline gamelogic/src/test/java/at/aau/se2/gamelogic/GameLogicTest.java:68]: Tests die wir nicht mehr brauchen können raus.

Machen wir neu, wenn wir die neuen Triggers/Actions in die Gamelogic implementieren.

**Review by raphaelseher — CHANGES_REQUESTED** [review]

File: gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/CardAction.java (hunk 3 above)
Comment (kuttnig) [reply gamelogic/src/main/java/at/aau/se2/gamelogic/models/cardactions/actions/CardAction.java:3]: True, der Grund dafür ist, dass wir in den Klassen DeployTrigger und OrderTrigger folgendes haben:

public class DeployTrigger {
  private ArrayList<CardAction> cardActions;

Die Klassen TargetUnitAction und TargetRowAction implementieren das Interface CardAction. Die Idee dabei ist das wir dann beim Iterieren über die cardActions mit "instanceof" checken, ob wir TargetUnit oder TargetRow haben.

**Comment by raphaelseher** [comment]
|File|Coverage [44.13%]|:green_apple:|
|:-|:-:|:-:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|81.22%|:green_apple:|
|[DeployTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FDeployTrigger.java)|66.67%|:green_apple:|
|[OrderTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FOrderTrigger.java)|52.94%|:green_apple:|
|[Card.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCard.java)|45.61%|:green_apple:|
|[JsonGenerator.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonGenerator.java)|0%|:green_apple:|
|[JsonConverter.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonConverter.java)|0%|:green_apple:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|0%|:green_apple:|
|[TargetRowAction.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetRowAction.java)|0%|:green_apple:|
|[TargetUnitAction.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetUnitAction.java)|0%|:green_apple:|
|[CardAction.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FCardAction.java)|0%|:green_apple:|
|[DetailedCardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardViewModel.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/35ce86a3725e859cf84047d244d536b372b5c0c5/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|

|Total Project Coverage|25.16%|:green_apple:|
|:-|:-:|:-:|

**Comment by raphaelseher** [comment]
|File|Coverage [33.98%]|:green_apple:|
|:-|:-:|:-:|
|[GameLogic.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FGameLogic.java)|81.15%|:green_apple:|
|[DeployTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FDeployTrigger.java)|66.67%|:green_apple:|
|[OrderTrigger.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Ftriggers%2FOrderTrigger.java)|52.94%|:green_apple:|
|[Card.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FCard.java)|45.61%|:green_apple:|
|[JsonGenerator.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonGenerator.java)|0%|:green_apple:|
|[JsonConverter.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Futil%2Fjson%2FJsonConverter.java)|0%|:green_apple:|
|[RowStatus.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2FRowStatus.java)|0%|:green_apple:|
|[TargetRowAction.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetRowAction.java)|0%|:green_apple:|
|[TargetUnitAction.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2Fmodels%2Fcardactions%2Factions%2FTargetUnitAction.java)|0%|:green_apple:|
|[CardAction.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/gamelogic%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgamelogic%2FCardAction.java)|0%|:green_apple:|
|[DetailedCardViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2Fviews%2Fdetailedcard%2FDetailedCardViewModel.java)|0%|:green_apple:|
|[MainGameActivity.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FMainGameActivity.java)|0%|:green_apple:|
|[StartFragment.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FStartFragment.java)|0%|:green_apple:|
|[GameDebugViewModel.java](https://github.com/SE2-Gwent/Gwent/blob/e3ffb9bd1cbce4ee7674e90b3f908f9ef074c4bb/app%2Fsrc%2Fmain%2Fjava%2Fat%2Faau%2Fse2%2Fgwent%2FGameDebugViewModel.java)|0%|:green_apple:|

|Total Project Coverage|25.12%|:green_apple:|
|:-|:-:|:-:|

**Review by raphaelseher — APPROVED** [review]

**Pull request merged**